libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -105,6 +111,15 @@ void Scheduler::init()
                      this);                     /* Thread parameter.    */
 #endif
 
+#ifndef HAL_NO_LED_THREAD
+    // setup the LED thread - this will call tasks at 1kHz
```

Comment:
this is starting the thread even on boards with no LEDs of this type. Most boards don't have them, so why not create dynamically to save memory/cpu?

Reply:
Fixed